multimodal/src/autogluon/multimodal/utils/export.py

```diff
@@ -193,14 +194,17 @@ def optimize_for_inference(
                 raise ValueError(f"unsupported column type: {col_type}")
         data = pd.DataFrame.from_dict(data_dict)
 
-        onnx_path = self.export_onnx(data=data, path=self.path, truncate_long_and_double=True)
-        onnx_module = OnnxModule(onnx_path, providers)
-        onnx_module.input_keys = self._model.input_keys
-        onnx_module.prefix = self._model.prefix
-        onnx_module.get_output_dict = self._model.get_output_dict
+        onnx_module = None
+        with tempfile.TemporaryDirectory() as tmpdirname:
```

Comment:
I think this method creates a temporary folder under the current directory, i.e. ./temp_abcd.
If the intention of this PR is to write artifacts to /tmp folder, I don't think this does the job.

On the other hand, the cloud module encountered such problem for timeseries predictor as well because it's writing artifacts to the endpoint while inferencing. I think a better way to deal with such an issue is to expose a parameter that allows the user to customize where to save the artifact instead of manually write it to /tmp folder

Reply:
good point. i've moved to utilize BytesIO instead.